src/backend/distributed/master/master_truncate.c

```diff
@@ -139,3 +144,75 @@ TruncateTaskList(Oid relationId)
 
 	return taskList;
 }
+
+
+/*
+ * truncate_local_table truncates the local records of a distributed table.
+ *
+ * The main advantage of this function is to truncate all local records after creating a
+ * distributed table, and prevent constraints from failing due to outdated local records.
+ */
+Datum
+truncate_local_table(PG_FUNCTION_ARGS)
+{
+	EnsureCoordinator();
+	Oid relationId = PG_GETARG_OID(0);
+
+	EnsureLocalTableCanBeTruncated(relationId);
+	EnsureTableOwner(relationId);
```

Comment:
ExecuteTruncate will explicitly check for TRUNCATE permissions, so we can omit this line.